docs/source/java/flight.rst

```diff
@@ -0,0 +1,302 @@
+    import org.apache.arrow.memory.RootAllocator;
+
+    import java.util.Iterator;
+    import java.util.concurrent.TimeUnit;
+
+    Location location = Location.forGrpcInsecure("0.0.0.0", 58609);
+
+    try(BufferAllocator allocator = new RootAllocator();
+        FlightClient tutorialFlightClient = FlightClient.builder(allocator, location).build()){
+
+        Iterator<Result> resultIterator = tutorialFlightClient.doAction(
+                new Action("test-timeout"),
+                CallOptions.timeout(2, TimeUnit.SECONDS)
+        );
+    } catch (Exception e) {
+        e.printStackTrace();
+    }
+
+On the server side, timeouts are transparent. For cancellation, the server needs to manually poll
+`setOnCancelHandler` or `isCancelled` to check if the client has cancelled the call, and if so,
+break out of any processing the server is currently doing.
+
+.. code-block:: Java
+
+    import org.apache.arrow.flight.FlightClient;
+    import org.apache.arrow.flight.FlightDescriptor;
+    import org.apache.arrow.flight.FlightStream;
+    import org.apache.arrow.flight.Location;
+    import org.apache.arrow.flight.Ticket;
+    import org.apache.arrow.memory.BufferAllocator;
+    import org.apache.arrow.memory.RootAllocator;
+    import java.nio.charset.StandardCharsets;
+
+    // Client
+    Location location = Location.forGrpcInsecure("0.0.0.0", 58609);
+    try(BufferAllocator allocator = new RootAllocator();
+        FlightClient tutorialFlightClient = FlightClient.builder(allocator, location).build()){
+        try(FlightStream flightStream = tutorialFlightClient.getStream(new Ticket(
+                FlightDescriptor.path("profiles").getPath().get(0).getBytes(StandardCharsets.UTF_8)))) {
+            // ...
+            flightStream.cancel("tutorial-cancel", new Exception("Testing cancellation opion!"));
+        }
+    } catch (Exception e) {
+        e.printStackTrace();
+    }
+    // Server
+    @Override
+    public void getStream(CallContext context, Ticket ticket, ServerStreamListener listener) {
+        // ...
+        listener.setOnCancelHandler(()->{
+                    // Implement logic to handle cancellation option
+                });
+    }
+
+Enabling TLS
+============
+
+TLS can be enabled when setting up a server by providing a certificate and key pair to `FlightServer.builder.useTls`.
+
+On the client side, use `FlightClient.builder.trustedCertificates`.
+
+Enabling Authentication
+=======================
+
+
+
+Custom Middleware
+=================
+
+Servers and clients support custom middleware (or interceptors) that are called on every
+request and can modify the request in a limited fashion. These can be implemented by implements
+`FlightServerMiddleware` and `FlightClientMiddleware` interface methods.
+
+Middleware are fairly limited, but they can add headers to a request/response.
+
+Example: Need to intercept Flight server headers at request and print values also for the response
```

Comment:
We don't really add examples for this in the other docs, this would be better off in the cookbook

Reply:
Deleted